parser/yaml_parser.F90

```diff
@@ -0,0 +1,429 @@
+   result(nkeys)
+   use iso_c_binding, only: c_char, c_int, c_bool
+   integer(kind=c_int), intent(in) :: file_id !< File id corresponding to the yaml file that was opened
+   integer(kind=c_int), intent(in) :: block_id !< Id of the parent_block
+   integer(kind=c_int) :: nkeys
+end function get_nkeys_binding
+
+!> @brief Private c function that gets the ids of the key-value pairs in a block (see yaml_parser_binding.c)
+subroutine get_key_ids_binding(file_id, block_id, key_ids) bind(c)
+   use iso_c_binding, only: c_char, c_int, c_bool
+   integer(kind=c_int), intent(in) :: file_id !< File id corresponding to the yaml file that was opened
+   integer(kind=c_int), intent(in) :: block_id !< Id of the parent_block
+   integer(kind=c_int), intent(inout) :: key_ids(*) !< Ids of the key-value pairs
+end subroutine get_key_ids_binding
+
+!> @brief Private c function that checks if a key_id is valid (see yaml_parser_binding.c)
+!! @return Flag indicating if the key_id is valid
+function is_valid_key_id(file_id, key_id) bind(c) &
+   result(is_valid)
+   use iso_c_binding, only: c_char, c_int, c_bool
+   integer(kind=c_int), intent(in) :: file_id !< File id corresponding to the yaml file that was opened
+   integer(kind=c_int), intent(in) :: key_id !< Key id to check if valid
+   logical(kind=c_bool) :: is_valid !< Flag indicating if the file_id is valid
+end function is_valid_key_id
+
+!> @brief Private c function that get the key from a key_id in a yaml file
+!! @return Name of the key obtained
+function get_key(file_id, key_id) bind(c) &
+   result(key_name)
+   use iso_c_binding, only: c_ptr, c_int, c_bool
+   integer(kind=c_int), intent(in) :: file_id !< File id corresponding to the yaml file that was opened
+   integer(kind=c_int), intent(in) :: key_id !< Id of the key-value pair of interest
+   type(c_ptr) :: key_name
+end function get_key
+
+!> @brief Private c function that get the value from a key_id in a yaml file
+!! @return String containing the value obtained
+function get_value(file_id, key_id) bind(c) &
+   result(key_value)
+   use iso_c_binding, only: c_ptr, c_int, c_bool
+   integer(kind=c_int), intent(in) :: file_id !< File id corresponding to the yaml file that was opened
+   integer(kind=c_int), intent(in) :: key_id !< Id of the key-value pair of interest
+   type(c_ptr) :: key_value
+end function get_value
+
+!> @brief Private c function that determines they value of a key in yaml_file (see yaml_parser_binding.c)
+!! @return c pointer with the value obtained
+function get_value_from_key_wrap(file_id, block_id, key_name, sucess) bind(c) &
+   result(key_value2)
+
+   use iso_c_binding, only: c_ptr, c_char, c_int, c_bool
+   integer(kind=c_int), intent(in) :: file_id !< File id of the yaml file to search
+   integer(kind=c_int), intent(in) :: block_id !< ID corresponding to the block you want the key for
+   character(kind=c_char), intent(in) :: key_name !< Name of the key you want the value for
```

Comment:
Does this need to be an array to be a c string?

Reply:
Done in https://github.com/NOAA-GFDL/FMS/pull/858/commits/8bef575cd06287affaa5cca472e856a36af7740c